homeassistant/components/transmission.py

```diff
@@ -0,0 +1,200 @@
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import discovery
+from homeassistant.helpers.event import track_time_interval
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_USERNAME,
+    CONF_MONITORED_CONDITIONS,
+)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['transmissionrpc==0.11']
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'transmission'
+DATA_TRANSMISSION = 'data_transmission'
+
+DEFAULT_NAME = 'Transmission'
+DEFAULT_PORT = 9091
+TURTLE_MODE = 'turtle_mode'
+
+SENSOR_TYPES = {
+    'active_torrents': ['Active Torrents', None],
+    'current_status': ['Status', None],
+    'download_speed': ['Down Speed', 'MB/s'],
+    'paused_torrents': ['Paused Torrents', None],
+    'total_torrents': ['Total Torrents', None],
+    'upload_speed': ['Up Speed', 'MB/s'],
+    'completed_torrents': ['Completed Torrents', None],
+    'started_torrents': ['Started Torrents', None],
+}
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(TURTLE_MODE, default=False): cv.boolean,
+        vol.Optional(CONF_MONITORED_CONDITIONS, default=['current_status']):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+SCAN_INTERVAL = timedelta(minutes=2)
+
+
+def setup(hass, config):
+    """Set up the Transmission Component."""
+    hass.data[DATA_TRANSMISSION] = TransmissionData(hass, config)
+    hass.data[DATA_TRANSMISSION].init_torrent_list()
+
+    def refresh(event_time):
+        """Get the latest data from Transmission."""
+        hass.data[DATA_TRANSMISSION].update()
+
+    track_time_interval(hass, refresh, SCAN_INTERVAL)
+
+    sensorconfig = {
+        'sensor_types': SENSOR_TYPES,
+        'sensors': config[DOMAIN][CONF_MONITORED_CONDITIONS],
+        'client_name': config[DOMAIN][CONF_NAME],
+        'component_name': DATA_TRANSMISSION}
+    discovery.load_platform(hass, 'sensor', DOMAIN, sensorconfig, config)
+
+    if config[DOMAIN][TURTLE_MODE] is True:
+        discovery.load_platform(hass, 'switch', DOMAIN, sensorconfig, config)
+    return True
+
+
+class TransmissionData:
+    """Get the latest data and update the states."""
+    
+    def __init__(self, hass, config):
+        """Initialize the Transmission RPC API."""
+        import transmissionrpc
+        from transmissionrpc.error import TransmissionError
+        try:
+            host = config[DOMAIN][CONF_HOST]
+            username = config[DOMAIN][CONF_USERNAME]
+            password = config[DOMAIN][CONF_PASSWORD]
+            port = config[DOMAIN][CONF_PORT]
+
+            api = transmissionrpc.Client(
```

Comment:
Move creating the api to `setup`. Pass `api` to `TransmissionData` when creating that.